lib/identity/auth.rb

```diff
@@ -332,13 +332,15 @@ def get_client_campaign(oauth_client_id)
     end
 
     def logout
-      @cookie.clear
-
-      url = if params[:url] && safe_redirect?(params[:url])
+      url = if !!@cookie.sso_entity &&  Config.sso_base_url
+        "#{Config.sso_base_url}/#{@cookie.sso_entity}"
```

Comment:
Can you extract this logic/construction into another method?
Also this:
https://github.com/heroku/identity/blob/redirect-sso-on-oauth/lib/identity/auth.rb#L22-L24